### PR TITLE
Added client-side checks for forwarding configs in advanced services

### LINKED ISSUE #307: Error messages for invalid forwarding config fields in the "Advanced Service" editor are not helpful
If a user tries to attach a config to a service with missing/invalid fields, the error message is difficult to interpret. It would be nice if the console could highlight/inform the user which fields are invalid.

See below screenshot for example:
Screen Shot 2024-04-09 at 5 37 31 PM

### PR DESCRIPTION
**Pull request by rgallettonf**
closes #307 
Will highlight missing/required fields and display a more user friendly error message growler.

See below screenshot for an example:

Screen Shot 2024-04-09 at 6 18 53 PM

### CODE REVIEW

**Comment by dovholuknf** [comment]
there's no way to turn this into a hover that makes it easier to understand and doesn't go away, is there? i might ask for too much -- but that would feel like a better UX

File: projects/ziti-console-lib/src/lib/features/projectable-forms/service/service-form.service.ts [hunk 1]
```diff
@@ -497,6 +504,30 @@ export class ServiceFormService {
         return data;
     }
 
+    validateConfigItems(items, parentType = 'object') {
+        let isValid = true;
+        items.forEach((item) => {
+            if (item.type === 'array') {
+                if (item?.component?.instance?.isValid) {
```
Comment (dovholuknf) [inline projects/ziti-console-lib/src/lib/features/projectable-forms/service/service-form.service.ts:511]: `item?.component?.instance?.isValid` is duplicated in this one else and also a bunch in projects/ziti-console-lib/src/lib/services/schema.service.ts...

can't this be made into a reusable function? feels like too much copy/paste code for my taste... :/

Comment (rgallettonf) [reply projects/ziti-console-lib/src/lib/features/projectable-forms/service/service-form.service.ts]: Updated to prevent copied/repeated code

**Comment by rgallettonf** [comment]
> there's no way to turn this into a hover that makes it easier to understand and doesn't go away, is there? i might ask for too much -- but that would feel like a better UX

Not sure I follow 100% but happy to discuss in more detail. Maybe we can iterate on it?

**Review by lfouts — APPROVED** [review]

**Pull request merged**